src/string_functions.rs

```diff
@@ -0,0 +1,2048 @@
+#[no_mangle]
+pub unsafe extern "C" fn REPLACE_EXT__STRING(
+    src_base: *const u8,
+    src_replacement: *const u8,
+    num_chars_to_replace: i32,
+    pos: i32,
+    dest: *mut u8,
+) -> i32 {
+    let mut dest = dest;
+    let nbase = EncodedCharsIter::decode(src_base).count();
+    if pos < 1 || pos > nbase as i32 {
+        panic!("Index out of bounds.")
+    }
+    // correct for 0-indexing
+    let pos = (pos - 1) as usize;
+    let nreplace = num_chars_to_replace as usize;
+
+    if nreplace + pos > nbase {
+        panic!(
+            r#"Cannot replace {} characters starting from index {}.
+            Index out of bounds.
+            "#,
+            nreplace,
+            pos + 1
+        )
+    }
+    EncodedCharsIter::decode(src_base)
+        .take(pos)
+        .chain(
+            EncodedCharsIter::decode(src_replacement)
+                .chain(EncodedCharsIter::decode(src_base).skip(pos + nreplace)),
+        )
+        .encode(&mut dest);
+
+    0
+}
+
+/// Replaces the given amount of characters in a string, starting
+/// from a specified location in the string, with another string and
+/// writes it to the destination buffer.
+/// Encoding: UTF-16
+///
+/// # Safety
+///
+/// Works on raw pointers, inherently unsafe.
+/// Will panic if trying to index outside of the array or trying
+/// to replace more characters than remaining.
+#[allow(non_snake_case)]
+#[no_mangle]
+pub unsafe extern "C" fn REPLACE_EXT__WSTRING(
+    src_base: *const u16,
+    src_replacement: *const u16,
+    num_chars_to_replace: i32,
+    pos: i32,
+    dest: *mut u16,
+) -> i32 {
+    let mut dest = dest;
+    let nbase = EncodedCharsIter::decode(src_base).count();
+    if pos < 1 || pos > nbase as i32 {
+        panic!("Index out of bounds.")
+    }
+    // correct for 0-indexing
+    let pos = (pos - 1) as usize;
+    let nreplace = num_chars_to_replace as usize;
+    if nreplace + pos > nbase {
+        panic!(
+            r#"Cannot replace {} characters starting from index {}.
+            Index out of bounds.
+            "#,
+            nreplace,
+            pos + 1
+        )
+    }
+    EncodedCharsIter::decode(src_base)
+        .take(pos)
+        .chain(
+            EncodedCharsIter::decode(src_replacement)
+                .chain(EncodedCharsIter::decode(src_base).skip(pos + nreplace)),
+        )
+        .encode(&mut dest);
+
+    0
+}
+
+/// Concatenates all given strings in the order in which they are given.
+/// Strings are passed as pointer of pointer to u8, where each pointer represents
+/// the starting address of each string. The amount of strings must be passed as
+/// argument.
+/// Encoding: UTF-8
+///
+/// # Safety
+///
+/// Works on raw pointers, inherently unsafe.
+/// Will panic if trying to index outside of the array or trying
+/// to replace more characters than remaining.
+#[allow(non_snake_case)]
+#[no_mangle]
+pub unsafe extern "C" fn CONCAT__STRING(argc: i32, argv: *const *const u8) -> Wrapper<[u8; 2048]> {
```

Comment:
I think it would make more sense to just call the extended method directly here

Reply:
I'm not sure if I made the change you wanted me to make here. If not, please just re-request changes and elaborate.